r/tests/testthat/test-dataset.R

```diff
@@ -318,6 +357,36 @@ test_that("compressed CSV dataset", {
   )
 })
 
+test_that("CSV dataset options", {
+  dst_dir <- make_temp_dir()
+  dst_file <- file.path(dst_dir, "data.csv")
+  df <- tibble(chr = letters[1:10])
+  write.csv(df, dst_file, row.names = FALSE, quote = FALSE)
+
+  format <- FileFormat$create("csv", skip_rows = 1)
+  ds <- open_dataset(dst_dir, format = format)
+
+  expect_equivalent(
+    ds %>%
+      select(string = a) %>%
+      collect(),
+    df1[-1,] %>%
+      select(string = chr)
+  )
+
+  format <- FileFormat$create("csv", column_names = c("foo"))
+  ds <- open_dataset(dst_dir, format = format)
+  expect_is(ds$format, "CsvFileFormat")
+  expect_is(ds$filesystem, "LocalFileSystem")
```

Comment:
We should have at least one of the tests testing that you can just pass the arguments in `...`

```suggestion
  ds <- open_dataset(dst_dir, format = "csv", column_names = c("foo"))
```